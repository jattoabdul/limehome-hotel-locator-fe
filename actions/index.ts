import axios from 'axios';
import {
  LOCATION_ONCHANGE,
  PROXIMITY_ONCHANGE,
  HOTELS_SET,
  HOTELS_CLEAR,
} from '../constants/actionTypes';

export const onChangeLocation = (location) => ({ type: LOCATION_ONCHANGE, location });

export const onChangeProximity = (proximity) => ({ type: PROXIMITY_ONCHANGE, proximity });

export const setHotelsAction = (hotels) => ({ type: HOTELS_SET, hotels });

export const clearHotels = () => ({ type: HOTELS_CLEAR });

export const setHotels = config => async (dispatch) => {
  let parameters = `key=${config.apiKey}&location=${config.location}&radius=${config.proximity}&type=lodging&keyword=hotel`;
  return await axios.get(`https://cors-anywhere.herokuapp.com/https://maps.googleapis.com/maps/api/place/nearbysearch/json?${parameters}`)
    .then(res => {
      const hotels = res.data;
      dispatch(setHotelsAction(hotels.results));
    })
};
