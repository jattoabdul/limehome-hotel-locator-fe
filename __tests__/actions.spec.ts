import configureMockStore from 'redux-mock-store'
import { onChangeProximity, onChangeLocation, setMapCenter } from '../actions';

const mockStore = configureMockStore()
const store = mockStore({})

describe('action creators', () => {
  it('creates LOCATION_ONCHANGE when changing location text is ongoing', () => {
    store.dispatch(
      onChangeLocation({
        value: '6.522500,3.321350'
      })
    );
    expect(store.getActions()).toMatchSnapshot();
  });
});
