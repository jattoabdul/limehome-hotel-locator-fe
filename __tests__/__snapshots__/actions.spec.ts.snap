// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action creators creates LOCATION_ONCHANGE when changing location text is ongoing 1`] = `
Array [
  Object {
    "location": Object {
      "value": "6.522500,3.321350",
    },
    "type": "LOCATION_ONCHANGE",
  },
]
`;
